### PR TITLE
EXT-80: customize horizon container for bsn plugin enablement [cherry-pick]

### PR DESCRIPTION
**Pull request by wolverineav**
Reviewer: @sarath-kumar 

 - cherry-pick

### CODE REVIEW

**Comment by bsn-bot3** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/openstack_horizon_bsn_pull_req/145/

File: rhel/horizon_container_workaround.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+FROM ${LOCAL_REGISTRY_ADDRESS}/openstack-horizon:${UPSTREAM_TAG}
+MAINTAINER Big Switch Networks Inc.
+LABEL name="rhosp12/openstack-horizon-bigswitch" vendor="Big Switch Networks Inc" version="11.0" release="1"
+
+# switch to root and install a custom RPM, etc.
+USER root
+COPY python-horizon-bsn*.rpm /tmp/
+COPY python-lockfile*.rpm /tmp/
+RUN rpm -ivh /tmp/*.rpm
+RUN cp /usr/lib/python2.7/site-packages/horizon_bsn/enabled/* /usr/lib/python2.7/site-packages/openstack_dashboard/local/enabled/
```
Comment (sarath-kumar) [inline rhel/horizon_container_workaround.sh]: Should this be 'COPY' instead of 'RUN cp'

File: rhel/horizon_container_workaround.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -eux
+
+CONTAINER_BUILD_DIR='openstack_horizon_bigswitch'
+
+# recreate worksapace for container build
```
Comment (sarath-kumar) [inline rhel/horizon_container_workaround.sh]: nit. typo 'workspace'

File: rhel/horizon_container_workaround.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+COPY python-horizon-bsn*.rpm /tmp/
+COPY python-lockfile*.rpm /tmp/
+RUN rpm -ivh /tmp/*.rpm
+RUN cp /usr/lib/python2.7/site-packages/horizon_bsn/enabled/* /usr/lib/python2.7/site-packages/openstack_dashboard/local/enabled/
+# switch the container back to the default user (NOT)
+# doing this has permission denied error during startup. skip it.
+# USER horizon
+EOF
+
+sudo docker build ./ -t latest_bsn_tag
```
Comment (sarath-kumar) [inline rhel/horizon_container_workaround.sh]: What exactly is 'latest_bsn_tag' here? Shouldn't this be bigswitch-horizon:latest ?

Comment (sarath-kumar) [reply rhel/horizon_container_workaround.sh]: I see you are using this as a temp docker img name here... why not #define on top and use it? And lets just call it openstack-horizon-bigswitch itself?

File: rhel/horizon_container_workaround.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+IMAGE_ID=`sudo docker images -q latest_bsn_tag`
+
+# tag latest build image and push to local registry
+sudo docker tag ${IMAGE_ID} ${LOCAL_REGISTRY_ADDRESS}/openstack-horizon-bigswitch:latest_bsn_tag
```
[the remark this answers is not in the document]
Comment (sarath-kumar) [reply rhel/horizon_container_workaround.sh]: or you can make it version - "12.1" (Pike + rev-version)

File: rhel/horizon_container_workaround.sh (hunk 1 above)
Comment (wolverineav) [reply rhel/horizon_container_workaround.sh]: The `COPY` instruction copies new files or directories from `<src>` and adds them to the filesystem of the container at the path `<dest>`.
Whereas `RUN cp` which I'm doing is to copy the enable files from the just installed package to its appropriate config location.

File: rhel/horizon_container_workaround.sh (hunk 3 above)
Comment (wolverineav) [reply rhel/horizon_container_workaround.sh]: yes, that's right - this is just a random unique identifier tag. i called it `latest_bsn_tag`.
i think the suggestion below is better - tag it based on version number: `12.1`. i'll update that. define a constant with the tag number and use it 👍

Comment (wolverineav) [reply rhel/horizon_container_workaround.sh]: i would just read it from the package name, but openstack's version number is 11.x.x for pike, whereas redhat's version is 12.x.x. which would just confuse the user. so i'll just keep it 12.0.1 for sake of ease and less confusion :)

**Comment by bsn-bot3** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/openstack_horizon_bsn_pull_req/147/

**Comment by wolverineav** [comment]
i presume this is okay to merge. let me know if there are any other comments. i'll open another PR.
getting a tarball ready for now :)

**Pull request merged**